webapp/packages/plugin-data-viewer/src/DatabaseDataModel/DatabaseDataActions.ts

```diff
@@ -52,6 +52,7 @@ export class DatabaseDataActions<TOptions, TResult extends IDatabaseDataResult>
     if (!action) {
       runInAction(() => {
         const allDeps = getDependingDataActions(Action).slice(1); // skip source argument
+        const resultIndexMap = new Map<string, number>(this.source.results.map((r, i) => [r.uniqueResultId, i]));
```

Comment:
Do you really think that:
1. Creating a new map
2. Creating a new Array
3. Creating N arrays
4. Getting an element from an array

Worth it to get a SINGLE element from this map?

This might be beneficial in case we get a lot of `get` from that map, but right now, it looks more like overhead.